spec/unit/parser_spec.rb

```diff
@@ -0,0 +1,255 @@
+require 'spec_helper'
+require 'iknow_params/parser'
+
+RSpec.describe IknowParams::Parser do
+  class Controller
+    attr_reader :params
+    def initialize(params)
+      @params = params
+    end
+  end
+
+  before { Controller.include IknowParams::Parser }
+
+  describe "#parse_param" do
+
+    let(:parsed) { Controller.new(@params).parse_param(@param, with: serializer) }
+
+    context "no serializer specified" do
+      let(:serializer) { nil }
+
+      it "returns the value" do
+        @params = {name: "foo"}
```

Comment:
You can't `let` inside an example. If you have a meaningful group that shares the variable, `let` is nicer for its lazy properties. Here, each example wants to change the value.